### PR TITLE
Implement Basic auth

### PR DESCRIPTION
**Pull request by jwgwalton**
Basic auth only on Create Update Delete endpoints as specified in documentation.

### CODE REVIEW

File: tests/views/test_classification_type_view.py [hunk 1]
```diff
@@ -4,24 +4,35 @@
...
+        response = self.client.post('/classificationtype/{}'.format(classification_type),
+                                    headers=self.get_auth_headers())
 
         # Then the classification type is successfully uploaded to the database
         self.assertStatus(response, 201)
 
+    def test_create_classification_type_without_basic_auth(self):
+        # when we upload a new classification type
+        classification_type = "LEGAL_BASIS"
+        response = self.client.post('/classificationtype/{}'.format(classification_type))
```
Comment (benjefferies) [inline tests/views/test_classification_type_view.py]: Could use `f'string interpolation'` for consistency

File: tests/views/test_template_view.py [hunk 2]
```diff
@@ -8,24 +8,34 @@ class TestTemplateView(TestClient):
...
 
-    def test_invalid_comms_template_upload_(self):
+    def test_create_comms_template_without_basic_auth(self):
+        # When a valid comms template is uploaded without correct basic auth
+        data = dict(id="cb0711c3-0ac8-41d3-ae0e-567e5ea1ef99", label="test data", type="EMAIL", uri="test-uri.com",
+                    classification={"GEOGRAPHY": "NI"})
+        response = self.client.post('/template/cb0711c3-0ac8-41d3-ae0e-567e5ea1ef99', content_type='application/json',
+                                    data=json.dumps(data))
+
+        # Then it isn't uploaded and we receive a 403 response
```
Comment (benjefferies) [inline tests/views/test_template_view.py]: `# Then it isn't uploaded and we receive a 403 response` -> `# Then it isn't uploaded and we receive a 401 response`

**Review by benjefferies — CHANGES_REQUESTED** [review]

File: tests/views/test_template_view.py (hunk 2 above)
Comment (jwgwalton) [reply tests/views/test_template_view.py]: @benjefferies not sure if you had any ideas on how to split up the test files as they're starting to get a bit long or is that alright?

File: tests/views/test_classification_type_view.py (hunk 1 above)
Comment (jwgwalton) [reply tests/views/test_classification_type_view.py]: Thought I'd got them all

Comment (jcox-dev) [reply tests/views/test_classification_type_view.py]: They sneak up on you ;)

File: tests/views/test_classification_type_view.py [hunk 3]
```diff
@@ -4,24 +4,32 @@
...
     def test_upload_existing_classification_type(self):
         # Given a ClassificationType exists in the database
         classification_type = "LEGAL_BASIS"
         self._create_classification_type(classification_type)
 
         # When the same classification type is uploaded
-        response = self.client.post(f'/classificationtype/{classification_type}')
+        response = self.client.post(f'/classificationtype/{classification_type}', headers=self.get_auth_headers())
 
         # Then the service returns a 400 response
```
Comment (benjefferies) [inline tests/views/test_classification_type_view.py]: 400 response -> 409 response?

File: tests/views/test_classification_type_view.py [hunk 4]
```diff
@@ -73,17 +81,28 @@ def test_delete_classification_type(self):
...
 
+    def test_delete_classification_type_without_basic_auth(self):
+        # Given there is an existing classification type in the database
+        classification_type = "LEGAL_BASIS"
+        self._create_classification_type(classification_type)
+
+        # When we delete the classification type
+        response = self.client.delete(f'/classificationtype/{classification_type}')
+
+        # Then the server returns a 200 response
```
Comment (benjefferies) [inline tests/views/test_classification_type_view.py]: 401?

File: tests/views/test_template_view.py [hunk 5]
```diff
@@ -33,20 +33,35 @@ def test_create_comms_template_with_no_classification_types_in_database(self):
...
+        # Given the classification type exists
+        self._create_classification_type("GEOGRAPHY")
+
+        # When a valid comms template is uploaded without correct basic auth
+        data = dict(id="cb0711c3-0ac8-41d3-ae0e-567e5ea1ef99", label="test data", type="EMAIL", uri="test-uri.com",
+                    classification={"GEOGRAPHY": "NI"})
+        response = self.client.post('/template/cb0711c3-0ac8-41d3-ae0e-567e5ea1ef99', content_type='application/json',
+                                    data=json.dumps(data))
+
+        # Then we receive a forbidden response
```
Comment (benjefferies) [inline tests/views/test_template_view.py]: forbidden -> unauthorised

**Review by benjefferies — APPROVED** [review]
Looks good just a few misleading comments

**Review by LJBabbage — APPROVED** [review]

**Pull request merged**